src/core/Impl/startOf.ts

```diff
@@ -33,9 +35,11 @@ export function startOfImpl(that: EsDay, unit: UnitType, reverse = false) {
       break
     case C.W:
     {
-      const dayOfWeek = callDateGetOrSet(result, 'Day', utc)
-      const diff = (dayOfWeek < 1 ? 7 : 0) + dayOfWeek - (reverse ? 6 : 0)
-      callDateGetOrSet(result, 'Date', utc, callDateGetOrSet(result, 'Date', utc) - diff)
+      // default start of week is Monday
+      // according to ISO 8601
+      const weekStart = that['$locale']().weekStart || 1
```

Comment:
I wouldn't use a magic number but define an appropriately named constant for the start of the week (e.g. dayOfWeekMonday =1)